server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2698,7 +2699,7 @@ protected boolean applyUserData(HypervisorType hyperVisorType, UserVm vm, Nic ni
     @Override
     @ActionEvent(eventType = EventTypes.EVENT_VM_START, eventDescription = "starting Vm", async = true)
     public UserVm startVirtualMachine(StartVMCmd cmd) throws ExecutionException, ConcurrentOperationException, ResourceUnavailableException, InsufficientCapacityException {
-        return startVirtualMachine(cmd.getId(), cmd.getHostId(), null, cmd.getDeploymentPlanner()).first();
+        return startVirtualMachine(cmd.getId(), cmd.getPodId(), cmd.getClusterId(), cmd.getHostId(), null, cmd.getDeploymentPlanner()).first();
```

Comment:
aren't these values supposed to be mutually exclusive? (pod, cluster, host)

Reply:
They can all be passed but only one of them will be selected and the following order host -> cluster -> pod